### DISCUSSION TITLE
The Black or white screen of nothing in TakeControl

### DISCUSSION THREAD

**Discussion by thisledome5** [amidaware/tacticalrmm#1864] [General]
Server : Ubuntu 22:04
Browser Chrome Edge Brave
RMM version 0.18.02

install method: standard
agent version: 2.7.0
Agent OS , Windows 10 Pro, 64 bit v22H2

So I have read through the few issues relating to this issue of a black or white screen connecting to the agent from the browser.  The history on the server shows a connect with takecontrol.,  but in every instance on every system with agent installed, The same result,
Either a  black or white screen pops up with black bar at top with RMM agent online and restart and recover buttons at top. thats it. nothing. So does anyone have any further advice?
I disabled the sync as one post suggest, same result. retarted server. etc. 

This package looks great, but this is not getting me anywhere. I would be happy to tier up and support if this works correctly

![Screenshot_12](https://github.com/amidaware/tacticalrmm/assets/45792300/178dc139-5250-4416-a7f6-69da2c4cc600)

**Comment by dinger1986** [comment]
If you go to mesh.yourdomain.tld can you connect there to the agent?

**Comment by thisledome5** [comment]
Hello. yes I can. That seems to work. so where is the problem?

**Comment by thisledome5** [comment]
However RDP doesn't seem to work. just goes to a black screen after I input credentials
But at least the mesh connect works as a temp measure. but why won't it connect directly from the tactical screen

**Comment by dinger1986** [comment]
Have you changed any settings under global settings for meshcentral?

**Comment by thisledome5** [comment]
No nothing for meshcentral. in fact I have not changed really much of anything at all.

**Comment by dinger1986** [comment]
see below (copied from discord) please give info on your setup.

Start your support requests with a good question, not a bad one. Read this:
http://www.catb.org/esr/faqs/smart-questions.html

TLDR

The RIGHT way to ask for help:
I have an established TRMM server (or: I have a new server), existing agents are working fine (can't get any agents to show up in TRMM/mesh). 
I updated from x version to y version of Tactical RMM (or: I haven't changed anything on my server config) today/last week/last month, now when I do xyz it looks like (insert screenshot).
When I look for (x), I see (y) when I'm expecting to see is (z). 

What's my next step? 

------

The WRONG way to ask for help:
xyz is not working. Server is fine.

------

Pre-filling the relevant answers to these questions will also help us help you:

Install type? (Standard/Docker) and amount of RAM on server
Did you deviate IN ANY WAY from these instructions? https://docs.tacticalrmm.com/install_server/
Did you deviate IN ANY WAY from these instructions? https://docs.tacticalrmm.com/install_docker/
Where is the server? (VPS/onprem)
New/old install? Rough age of TRMM server (days/weeks/months)?

Server Install Specific questions:
What OS/version is TRMM installed on.
Using a real domain?
Did letsencrypt finalize and work
Have you looked at the troubleshooting steps to see if something there is appropriate to you situation? Test DNS from an agent etc. https://docs.tacticalrmm.com/troubleshooting/
Run the Server Troubleshooting Script and DM the person (from @Support ) that is working with you https://docs.tacticalrmm.com/troubleshooting/#server-troubleshooting-script
What kind of ssl certs? Let's Encrypt, or purchased (you're not trying to make self-signed work right?)
Check Expiry date of your certificates in the browser (at https://rmm.example.com/ )

Network Troubleshooting
Are you using a proxy?
Are you a wizard? See https://docs.tacticalrmm.com/unsupported_guidelines/
If so, what's in the network between agent and server?

Agent Troubleshooting
Is there ANY 3rd party Antivirus installed on the computer?
Is there any network based filtering/AV filtering?

Asking yourself questions like: When did it go from working to broken? What changed during that time? This will help you.

**Comment by thisledome5** [comment]
really. thats your advice to a new person. I did follow the guideliness. I gave you the specs as laid out before I started writing.
Not the right way to help a newb to this place or this software mate.
Again no no no and no.
I did not deviate from the instructions one little bit. infact i thought I was at fault so i scrubbed the server 3 times are did it all over again just to be sure. 3 times. 
There is no 3rd party AV or network filtering 
there is no proxy
no wizard.  just the correct script instruction as per tactical rmm site for install
yes I did red through troubleshooting and also previous answers on the issues in github.  nothing changed , as it did not work from the install.certificates are fresh,   

you seem to be missing the fact that this is a fresh install. first time. yesterday. as yet. the remote connection has not worked from the panel. 
And the server is a hosted VPS in the cloud that I installed. comes wth ubuntu 22:04 installed at my request. all updates done. nothing else installed etc.
yes I use a real domain. yes lets encrypt finished. everything else finished right down the script with no errors. ok

I can't explain anymore. I have seena  few other people have this issue. so far no real fix or advice from anyone on how to mitigate it. ok

**Reply by dinger1986** [reply]
As I said I literally copied and pasted that from discord and I was trying to help, you would not believe the number of people who have been helped over the years and then turn round and say (after many questions) that they are running a proxy or something else unsupported which stops things working. Only trying to help get you up and running.

Thanks for providing the feedback.

So currently sync is switched off? If it's on and you log into mesh are the users created?

Do you have permissions setup yet on sites etc?

**Reply by thisledome5** [reply]
Hi, look I totally get that. and I probably jumped on it
So if I go to mesh. and login, the accounts are all there. and I can remote access every unit. although if I select RDP it doesn;t login, just another black screen. but meshcentral works and all the connected accounts from TRMM are there and working. it just won't work if I use it from TRMM
I just ran the netconnection test from powershell and connection is true
ye  have tried it both ways with Sync. off and on. makes no different

So the issue must be between the TRMM and meshcentral if meshcentral is working

I appreciate the help and I totally get the issues.  I am just  a newb to this software.. I have used Meshcentral before.

**Comment by silversword411** [comment]
No connect button means your mesh server has problems.

What's the output of the mesh troubleshooting script?

https://docs.tacticalrmm.com/troubleshooting/#agents-not-installing-or-updating

**Comment by silversword411** [comment]
If mesh agent is not connected to server there are no connect buttons.
Trying that mesh repair button isn't going to hurt either

**Comment by thisledome5** [comment]
ok so funny thing happened today. The remote sessions started working. I have not changed anything at all. Wasn't working yesterday. Today it works. so the take control mesh central remote works.
However I need to change the access port for the RDP client on the screen and could not see anywhere in the docmation that advises this particular need. Does anyone know what that might be or if it is possible. as all my clients are not on the standard 3389 due to increased attempts to infiltrate

**Reply by dinger1986** [reply]
ok, so probably was a dns issue which is now resolving.

Glad its working, do you open up rdp to the world?

**Reply by thisledome5** [reply]
thanks. I don;t know how it was a DNS issue. but hey who knows  maybe it hadn't fullly resolved somewhere

no, RDP is only enabled on the local pc. which is then connected to a secure SDWAN. only the computers on that WAN can access those IP's and yes the server with TRMM installed is also connected to the SDWAN

**Comment by tscheiber5** [comment]
Hi,

If you right-click the RDP-Connect button you can enter an alternative port.
![image](https://github.com/amidaware/tacticalrmm/assets/26470855/7307c4d7-66f9-453b-af16-97c21581e8d2)

**Reply by thisledome5** [reply]
Ok so still no luck on any RDP connection. through takecontrol.
RDP works fine directly to the servers so not sure why it does nothing but a black screen. There is a disconnect button top left border but it doesn;t seem to be connected at all

**Comment by CaymanS0** [comment]
@thisledome5 did you solved it?

**Reply by silversword411** [reply]
There's multiple things that can cause the remote control to not have the button. 

You should open your own discussion after reviewing troubleshooting from docs and posting the steps you've taken to test and described your setup

**Comment by JReming85** [comment]
Chances are the SSL cert your using wasnt added to the .env or docker config.

You can either update the meshcentral config to point directly to the mesh.domain.com url to check for the correct SSL cert, or update the env file.

The mesh.domain.com ssl cert has to match the cert meshcentral is using, otherwise the agent get refused. That or look to see if there is an option to disable SSL checking
